src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -32,6 +32,8 @@ public abstract class Element : IDisposable, IGraphicItem, IFormattable
         /// <param name="init"></param>
         protected void SafeInit(Action init)
         {
+            TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
```

Comment:
@Randy-Ma This is a good point. I'll double check that the task is opening a transaction, but we will get into trouble if another operation closes the transaction before we get here.